### ISSUE TITLE
[Bug] AWQ的实现和论文差距过大

### ISSUE THREAD

**Issue by matrix-yang** [InternLM/lmdeploy#381]
### Checklist

- [X] 1. I have searched related issues but cannot get the expected help.
- [X] 2. The bug has not been fixed in the latest version.

### Describe the bug

关于scale的计算：
lmdeploy/lite/quantization/awq.py
#66行
#act_scales  没有取对应权重的act_scales ，在上一级方法中取的 a_name = [f'{l_name}.{n}' for n in fc_names][0]
act_scales = act_scales.to(device=device, dtype=dtype)
concat_w = torch.cat([fc.weight for fc in fcs], dim=0)
w_scales = get_weight_scale(concat_w, group_size)

scales = (act_scales.pow(alpha) /
          w_scales.pow(1 - alpha)).clamp(min=1e-4).to(device).to(dtype)
#scales 的计算，论文中像是没有这一步
scales = scales / (scales.max() * scales.min()).sqrt()

scales 的计算和论文中差距甚远（没有对应act_scales）：
![image](https://github.com/InternLM/lmdeploy/assets/30305908/792e99d1-57fb-4438-889b-eba24432a4dd)
alpha和beta是通过grid search 获得最优，而lmdeploy 是设置alpha和beta都为0.5

会不是这个原因导致AWQ没办法量化这些模型。
https://github.com/InternLM/lmdeploy/issues/349
https://github.com/InternLM/lmdeploy/issues/302

### Reproduction

NO

### Error traceback

```Shell
NO
```

**Comment by pppppM** [comment]
> #act_scales 没有取对应权重的act_scales ，在上一级方法中取的 a_name = [f'{l_name}.{n}' for n in fc_names][0]
这个问题不是很理解，`smooth_fc_fcs(fc, fcs, a_scales[a_name], group_size)` 中 a_scales[a_name] 不是对应的 act_scales 么？

> #scales 的计算，论文中像是没有这一步
scales = scales / (scales.max() * scales.min()).sqrt()

这个操作是根据作者开源代码实现的，`scales = scales / (scales.max() * scales.min()).sqrt()` 

> alpha和beta是通过grid search 获得最优，而lmdeploy 是设置alpha和beta都为0.5
Autotune 的功能正在开发中，官方代码的 autotune 实现方式对显存要求比较高，LMDeploy 正在设计最低成本的 Autotune 实现方案

**Comment by matrix-yang** [comment]
#act_scales 没有取对应权重的act_scales ，在上一级方法中取的 a_name = [f'{l_name}.{n}' for n in fc_names][0]
fc_names的取值有['self_attn.k_proj','self_attn.q_proj','self_attn.v_proj'] 分别对应了fcs中的q,k，v
a_name = [f'{l_name}.{n}' for n in fc_names][0]
但是a_name只取了第0个
![image](https://github.com/InternLM/lmdeploy/assets/30305908/c8442d9b-1c6d-4810-b3a4-390f3c84448a)
上一级的实现。

另外的两个 操作，感谢解惑。

**Comment by pppppM** [comment]
Norm 和 FCS 的映射，是指 Norm 的输出是哪些 FC 的输入，所以这里每个 FC 的输入都是一样的，取第 0 个或者取 1、2 都是一样的

**Comment by matrix-yang** [comment]
感谢解惑，我现在了解了scale 的计算情况了，除了这一点（Autotune 的功能正在开发中）：
1.alpha和beta是通过grid search 获得最优，而lmdeploy 是设置alpha和beta都为0.5
我先尝试挑一下alpha和beta，看能不能解决llama2 70B的量化问题。

**Comment by pppppM** [comment]
Autotune 就是你说的 grid search

**Comment by nexa123** [comment]
通过相同的推理参数，我使用llm-awq的脚本和lmdeploy生成的结果也很不一样。lmdeploy的awq实现输出有和llm-awq对齐吗

**Comment by lvhan028** [comment]
kernel的实现方式和 llm-awq 不一样。结果不一样也是正常的。重要的是，模型生成的结果是否正常、流畅。下周，我们争取把lmdeploy对接到opencompass上，给出 lmdeploy w4a16 在数据集上的精度测试结果
